typescript/src/WorkQueue.ts

```diff
@@ -0,0 +1,270 @@
+    pipeline.set(this.itemDataKey.of(itemId), item.data);
+    /**
+     * Then add the id to the work queue
+     */
+    pipeline.lpush(this.mainQueueKey, itemId);
+    await pipeline.exec();
+  }
+
+  /**
+   * Add an item to the work queue.
+   * This creates a pipeline and executes it on the database.
+   *
+   * @param {Redis} db The Redis Connection.
+   * @param item The item that will be executed using the method addItemToPipeline.
+   */
+  addItem(db: Redis, item: typeof Item): void {
+    const pipeline = db.pipeline() as unknown as Pipeline;
+    this.addItemToPipeline(pipeline, item);
+  }
+
+  /**
+   * This is used to get the lenght of the Main Queue.
+   *
+   * @param {Redis} db The Redis Connection.
+   * @returns {Promise<number>} Return the length of the work queue (not including items being processed, see `WorkQueue.processing()`).
+   */
+  queueLen(db: Redis): Promise<number> {
+    return db.llen(this.mainQueueKey);
+  }
+
+  /**
+   * This is used to get the lenght of the Processing Queue.
+   *
+   * @param {Redis} db The Redis Connection.
+   * @returns {Promise<number>} The number of items being processed.
+   */
+  processing(db: Redis): Promise<number> {
+    return db.llen(this.processingKey);
+  }
+
+  /**
+   * This method can be used to check if a Lease Exists or not for a itemId.
+   *
+   * @param {Redis} db The Redis Connection.
+   * @param {string} itemId The itemId of the item you want to check if it has a lease.
+   * @returns {Promise<boolean>}
+   */
+  async leaseExists(db: Redis, itemId: string): Promise<boolean> {
+    return await db.exists(this.leaseKey.of(itemId)).then(exists => {
+      return exists !== 0;
+    });
+  }
+
+  /**
+   * Request a work lease from the work queue. This should be called by a worker to get work to complete.
+   * When completed, the `complete` method should be called.
+   *
+   * If `block` is true, the function will return either when a job is leased or after `timeout` seconds if `timeout` isn't 0.
+   * If the job is not completed before the end of `leaseDuration`, another worker may pick up the same job.
+   * It is not a problem if a job is marked as `done` more than once.
+   *
+   * If you haven't already, it's worth reading the documentation on leasing items:
+   * https://github.com/MeVitae/redis-work-queue/blob/main/README.md#leasing-an-item
+   *
+   * @param {Redis} db The Redis Connection.
+   * @param {number} leaseSecs The number of seconds that the lease should hold.
+   * @param {boolean} block Is a block or not, default is true.
+   * @param {number} timeout The number of seconds the lease will time out at.
+   * @returns {Promise<typeof Item>} Returns a new lease Item.
+   *
+   * Process:
+   * First, to get an item, we try to move an item from the main queue to the processing list.
+   * Then we setup the lease item.
+   */
+  async lease(
+    db: Redis,
+    leaseSecs: number,
+    block = true,
+    timeout = 1
+  ): Promise<typeof Item> {
```

Comment:
(I don't know a whole lot of typescript so there's probably a good reason I'm not aware of) Is there a reason this is `typeof Item` and not just `Item`?

Reply:
Typeof gets the type of the class Item where if we would use Item directly it would be a instance of Item.

Reply:
You are right, the way I was importing Item did not allow me to use the class as a type.

I fixed it, will push after.